bundle/LegacyMapper/SiteAccess.php

```diff
@@ -90,6 +90,24 @@ public function onBuildKernelWebHandler( PreBuildKernelWebHandlerEvent $event )
             $uriPart = array_diff( $aPathinfo, $aSemanticPathinfo, $this->getFragmentPathItems() );
         }
 
+        if ($siteAccess->matchingType == 'compound:logicalAnd(Map\Host, Map\URI)')
+        {
+            $subMatchers = $siteAccess->matcher->getSubMatchers();
+            if ($subMatchers)
+            {
+                foreach ($subMatchers as $key => $subMatcher)
+                {
+                    if ($key == 'Map\URI')
```

Comment:
Why are you re-checking it here? You could directly access to the right matchers.

Furthermore, if you don't have any sub matchers, you should raise an exception, not only apply the following code if `$subMatchers` isn't empty.

```php
$subMatchers = $siteAccess->matcher->getSubMatchers();
if ( !$subMatchers )
{
    // Throw an exception as the situation is not normal at all
}

// Continue normally
```